waku/waku_sync/raw_bindings.nim

```diff
@@ -63,67 +55,42 @@ proc fromBytes(T: type String, bytes: seq[byte]): T =
 
-type
-  Storage* {.importcpp: "negentropy::storage::BTreeMem", header: NEGENTROPY_HEADER, byref.} = object
+proc Storage(db_path:cstring, name: cstring) :pointer{. header: NEGENTROPY_HEADER, importc: "storage_new".}
 
-# https://github.com/hoytech/negentropy/blob/6e1e6083b985adcdce616b6bb57b6ce2d1a48ec1/cpp/negentropy/storage/btree/core.h#L163
-proc raw_insert(this: Storage, timestamp: clong, id: String) {.importcpp: "negentropy::storage::btree::insert".}
+#[ # https://github.com/hoytech/negentropy/blob/6e1e6083b985adcdce616b6bb57b6ce2d1a48ec1/cpp/negentropy/storage/btree/core.h#L163
+proc raw_insert(this: Storage, timestamp: clong, id: String) {.importc: "negentropy::storage::btree::insert".}
 
 # https://github.com/hoytech/negentropy/blob/6e1e6083b985adcdce616b6bb57b6ce2d1a48ec1/cpp/negentropy/storage/btree/core.h#L300
-proc raw_erase(this: Storage, timestamp: clong, id: String) {.importcpp: "negentropy::storage::btree::erase".}
-
+proc raw_erase(this: Storage, timestamp: clong, id: String) {.importc: "negentropy::storage::btree::erase".}
+ ]#
 ### Negentropy ###
 
-type
-  Negentropy* {.importCpp: "negentropy::Negentropy", header: NEGENTROPY_HEADER, byref.} = object
-
 # https://github.com/hoytech/negentropy/blob/6e1e6083b985adcdce616b6bb57b6ce2d1a48ec1/cpp/negentropy.h#L42
-proc constructNegentropy(storage: Storage, frameSizeLimit: culong): Negentropy {.importcpp: "negentropy::Negentropy(@)", constructor.}
+proc constructNegentropy(storage: pointer, frameSizeLimit: BiggestUInt): pointer {.importc: "negentropy_new".}
 
 # https://github.com/hoytech/negentropy/blob/6e1e6083b985adcdce616b6bb57b6ce2d1a48ec1/cpp/negentropy.h#L46
-proc raw_initiate(this: Negentropy): String {.importcpp: "negentropy::initiate".}
-
+proc raw_initiate(negentropy: pointer): cstring {.importc: "negentropy_initiate".}
+#[ 
 # https://github.com/hoytech/negentropy/blob/6e1e6083b985adcdce616b6bb57b6ce2d1a48ec1/cpp/negentropy.h#L58
-proc raw_setInitiator(this: Negentropy) {.importcpp: "negentropy::setInitiator".}
+proc raw_setInitiator(this: Negentropy) {.importc: "negentropy::setInitiator".}
 
 # https://github.com/hoytech/negentropy/blob/6e1e6083b985adcdce616b6bb57b6ce2d1a48ec1/cpp/negentropy.h#L62
-proc raw_reconcile(this: Negentropy, query: String): String {.importcpp: "negentropy::reconcile".}
+proc raw_reconcile(this: Negentropy, query: String): String {.importc: "negentropy::reconcile".}
 
 # https://github.com/hoytech/negentropy/blob/6e1e6083b985adcdce616b6bb57b6ce2d1a48ec1/cpp/negentropy.h#L69
-proc raw_reconcile(this: Negentropy, query: String, haveIds: var Vector[String], needIds: var Vector[String]): String {.importcpp: "negentropy::reconcile".}
-
+proc raw_reconcile(this: Negentropy, query: String, haveIds: var Vector[String], needIds: var Vector[String]): String {.importc: "negentropy::reconcile".}
+ ]#
 ### Wrappings ###
 
-proc new*(T: type Storage): T =
-  let storage = Storage()
+proc new_storage*(): pointer =
+  let storage = Storage("", "")
```

Comment:
I don't want to expose the unsafe internals.

The procs marked as `*` are public and should not contain any pointers or cstring. We keep C on the private side and expose only pure Nim on the public side.

Unnecessary until the polishing phase but FYI.

Reply:
Oh, was not aware of that in nim. Makes sense, will change this.